src/HospitalLibrary/Shared/Validators/TimeIntervalValidationService.cs

```diff
@@ -18,7 +18,19 @@ public TimeIntervalValidationService(IUnitOfWork unitOfWork)
         {
             _unitOfWork = unitOfWork;
         }
-        
+
+        public async Task<bool> IsIntervalOverlapingWithDoctorAppointments(int doctorId,TimeInterval possibleTimeInterval)
+        {
+            IEnumerable<TimeInterval> doctorsAppointmentsTimeIntervals = (await _unitOfWork.AppointmentRepository
+                .GetAllDoctorTakenIntervalsForDate(doctorId,possibleTimeInterval.Start));
+            foreach (TimeInterval timeInterval in doctorsAppointmentsTimeIntervals)
```

Comment:
I think there is a LINQ expression that can write this much shorter.
Something like: doctorAppointmentTimeIntervals.Any(app => possibleTimeInterval.IsOverlaping(app));